04_001/backend/src/routes.ts

```diff
@@ -0,0 +1,142 @@
+import path from "path";
+import { Router, Request, Response } from "express";
+import { createMessage, getPostsData, setPostData } from "./utils";
+import { httpStatus } from "./utils";
+import { PostData, ResponseDataType } from "@common/types";
+import { ServerError } from "./core";
+
+// ----
+const { OK, BAD_REQUEST, INTERNAL_SERVER_ERROR } = httpStatus;
+const jsonFileName = path.resolve("src/data", "posts.json");
+const cacheMap: Map<string, ResponseDataType<PostData | PostData[]>> = new Map();
+// ----
+
+const router = Router();
+
+// '/api/post/:id': 글 가져오기 (id 있을 때는 해당하는 데이터만 가져옴)
+router.get("/post/:id?", async (req: Request, res: Response) => {
+  try {
+    const routeName = "api: posts";
+    const id = req.params.id;
+    const isId = Boolean(id);
+    const POST_KEY = isId ? `posts_${id}` : `posts`;
+
+    // 캐싱되어있는지 확인
+    if (cacheMap.has(POST_KEY)) {
+      const cache = cacheMap.get(POST_KEY);
+      return res.status(OK).json(cache);
+    }
+
+    // JSON 파일 가져오기
+    const { isError, result: postData } = getPostsData({ jsonFileName, id: isId ? +id : undefined });
+    if (isError || !postData)
+      throw new ServerError(createMessage({ type: "NOT_FOUND_POST_DATA", name: routeName }), BAD_REQUEST);
+
+    // 캐싱 업데이트 & 반환
+    const message = createMessage({ type: "SUCCESS" });
+    const result: ResponseDataType<PostData | PostData[]> = { statusCode: OK, message, data: postData };
+    cacheMap.set(POST_KEY, result);
+    return res.status(OK).json(result);
+  } catch (e) {
+    console.error(e);
+    const message = (e as Error).message;
+    const result = { statusCode: res.statusCode, message, data: null };
+    return res.status(res.statusCode).json(result);
+  }
+});
+
+// '/api/write': 글 작성
+router.post("/write", async (req: Request, res: Response) => {
+  try {
+    const routeName = "api: write";
+    if (!req.body) throw new ServerError(createMessage({ type: "NO_DATA_SENT", name: routeName }), BAD_REQUEST);
```

Comment:
```suggestion
if (!req.body) throw new PostWriteError();
```
이렇게 표현할 수 있도록 Error 객체를 하나 만들어주면 어떨까요? 뒤에 있는 코드들도 동일합니다!